### PR TITLE
fix: start of SDF will not send measurement

### PR DESCRIPTION
**Pull request by Roy-Hu**

### CODE REVIEW

**Review by afcidk — APPROVED** [review]
TS 29.244 5.2.2.1

The UP function sends a usage report without performing network resources usage measurements when being requested to detect and report the start of an SDF or application traffic.

File: src/genl/genl_report.c [hunk 1]
```diff
@@ -336,19 +336,23 @@ int gtp5g_genl_fill_multi_usage_reports(struct sk_buff *skb, u32 snd_portid, u32
...
                 0,
                 urr->start_time,
                 urr->end_time,
                 urr->seid
         };
 
-    memset(&urr->bytes, 0, sizeof(struct VolumeMeasurement));
+    if (report_vol){
+        report->volmeasurement = urr->bytes;
+        memset(&urr->bytes, 0, sizeof(struct VolumeMeasurement));
```
Comment (tim-ywliu) [inline src/genl/genl_report.c]: this line should move out `if (report_vol)`

File: src/genl/genl_report.c [hunk 2]
```diff
@@ -336,19 +336,21 @@ int gtp5g_genl_fill_multi_usage_reports(struct sk_buff *skb, u32 snd_portid, u32
...
                 0,
-                urr->bytes,
+                (struct VolumeMeasurement) {0}  ,
                 0,
                 urr->start_time,
                 urr->end_time,
                 urr->seid
         };
 
+    report->volmeasurement = urr->bytes;
```
Comment (tim-ywliu) [inline src/genl/genl_report.c]: why remove `if (report_vol)`?

**Comment by Roy-Hu** [comment]
TS 29.244 5.4.11
creating a URR with the Reporting Trigger IE set to detect the start and/or stop of Traffic; 
- the CP function may include a zero quota together with a FAR ID for Quota Action IE in the URR to instruct 
the UP function to drop or buffer the packets pertaining to the detected application traffic before the quota is 
granted in the subsequent PFCP Session Modification Request message

Currently, the FAR ID for Quota Action IE is not implemented. Therefore,  instead of a FAR ID for Quota Action IE that indicates the action while no quota is granted, a predefined drop action will be performed

**Comment by Bunos07** [comment]
Fix

**Review by Bunos07 — APPROVED** [review]
Plss

**Review by Bunos07 — APPROVED** [review]

**Review by Bunos07 — APPROVED** [review]

**Review by Bunos07 — APPROVED** [review]

**Review by Bunos07 — APPROVED** [review]
Fix

File: src/genl/genl_report.c
Comment (tim-ywliu) [inline src/genl/genl_report.c:1]: rollback this file

File: src/gtpu/encap.c [hunk 3]
```diff
@@ -593,7 +593,7 @@ int check_urr(struct pdr *pdr, struct far *far, u64 vol, u64 vol_mbqe, bool upli
                 if (urr->trigger == 0) {
                     GTP5G_ERR(pdr->dev, "no supported trigger(%u) in URR(%u) and related to PDR(%u)",
                         urr->trigger, urr->id, pdr->id);
-                    ret = 0;
+                    ret = 0;
```
Comment (tim-ywliu) [inline src/gtpu/encap.c]: remove redundant space

**Review by tim-ywliu — APPROVED** [review]

**Pull request merged**